### PR TITLE
evidence: check evidence is pending before validating evidence

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

In order to make block validation quicker, evidence will be checked whether it is already pending in the evidence pool before it is verified. (if it is pending it has already been verified before)

Closes: #4728

---

For contributor use:

- [ ] Wrote tests
- [ ] Updated CHANGELOG_PENDING.md
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [x] Applied Appropriate Labels

### CODE REVIEW

File: state/validation.go [hunk 1]
```diff
@@ -132,12 +132,17 @@ func validateBlock(evidencePool EvidencePool, stateDB dbm.DB, state State, block
 
 	// Validate all evidence.
 	for _, ev := range block.Evidence.Evidence {
+		if evidencePool != nil {
+			if evidencePool.IsCommitted(ev) {
+				return types.NewErrEvidenceInvalid(ev, errors.New("evidence was already committed"))
+			}
+			if evidencePool.IsPending(ev) {
+				return nil
```
Comment (melekes) [inline state/validation.go:140]: not continue?

Comment (cmwaters) [reply state/validation.go:140]: yeah good catch, for some reason I forgot about the for loop

Comment (melekes) [reply state/validation.go:140]: this would be a signal for me that the unit test is needed 🧪

Comment (cmwaters) [reply state/validation.go:140]: yeah the current tests are quite basic atm i.e. just one evidence - I can further expand.

Comment (cmwaters) [reply state/validation.go:140]: Do you want me to consolidate the mocks into one and try use mockery?

Comment (melekes) [reply state/validation.go:140]: try to use mockery

File: state/validation_test.go [hunk 2]
```diff
@@ -260,15 +261,70 @@ func TestValidateFailBlockOnCommittedEvidence(t *testing.T) {
...
+	// add evidence that hasn't seen before
+	ev := types.NewMockEvidence(height, time.Now(), addr2)
+	block.Evidence.Evidence = []types.Evidence{pendingEv, ev}
+	block.EvidenceHash = block.Evidence.Hash()
+	err := blockExec.ValidateBlock(state, block)
+
+	require.NoError(t, err)
+}
+
+// TODO: prevent committing duplicate votes
```
Comment (melekes) [inline state/validation_test.go:307]: is there an issue for this?

Comment (cmwaters) [reply state/validation_test.go:307]: #4756

**Comment by cmwaters** [comment]
Do you know why this is failing? it seems like something wrong with the cli

**Pull request merged**